mp/src/game/client/hl2/hud_credits.cpp

```diff
@@ -137,8 +120,50 @@ class CHudCredits : public CHudElement, public Panel
     float m_flCreditsPixelHeight;
 
     HFont m_hTFont;
+
+    int m_iMusicGUID;
+
+    ConVarRef sv_unlockedchapters;
 };
 
+using namespace vgui;
+
+static CHudCredits *g_pCreditsDialog = nullptr;
+
+//-----------------------------------------------------------------------------
+// Purpose: Constructor
+//-----------------------------------------------------------------------------
+CHudCredits::CHudCredits() : BaseClass(nullptr, "HudCredits"), sv_unlockedchapters("sv_unlockedchapters")
+{
+    g_pCreditsDialog = this;
+
+    SetParent(enginevgui->GetPanel(PANEL_GAMEUIDLL));
+    SetProportional(true);
+
+    // for mouse speedup
+    SetMouseInputEnabled(true);
+    SetKeyBoardInputEnabled(true);
+
+    SetMoveable(false);
+    SetSizeable(false);
+    SetDeleteSelfOnClose(false);
+    SetMinimizeToSysTrayButtonVisible(false);
+    SetSysMenu(nullptr);
+
+    SetTitleBarVisible(false);
+    SetMenuButtonVisible(false);
+    SetMenuButtonResponsive(false);
+    SetMinimizeButtonVisible(false);
+    SetMaximizeButtonVisible(false);
+    SetCloseButtonVisible(false);
+
+    int iWidth, iTall;
+    surface()->GetScreenSize(iWidth, iTall);
+    SetBounds(0, 0, iWidth, iTall);
```

Comment:
Override `PerformLayout` and do this there, as this will help with resolution changes.

Reply:
Originally had it there, though the credits preparation, which calls `GetWide()` and `GetTall()` happens before the `PerformLayout` causing some buggy behavior. Might be fine when I create that separate resource file as the wide and tall will be loaded in the constructor.

Reply:
You can do it in both. And regardless of implementation, ensure the panel is full screen upon resolution changes.